plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -48,15 +66,23 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
    std::optional<state_history_log> trace_log;
    std::optional<state_history_log> chain_state_log;
    bool                             trace_debug_mode = false;
-   bool                             stopping         = false;
+   std::atomic<bool>                stopping         = false;
    std::optional<scoped_connection> applied_transaction_connection;
    std::optional<scoped_connection> block_start_connection;
    std::optional<scoped_connection> accepted_block_connection;
-   string                           endpoint_address = "0.0.0.0";
+   string                           endpoint_address;
```

Comment:
Use a single string for the endpoint specification, either IP address or unix socket path.

Reply:
That sounds like a significant change to behavior: the my original implementation deliberately allowed both TCP & Unix socket to be available simultaneously.

Reply:
Oh I didn't realize they were intended to operate simultaneously.  In that case changes made to address my comments about consolidating endpoint strings and the acceptors in a variant should be backed out.

Reply:
I knew that they can be active at the same time, but missed it during this refactoring... I adapted new pattern to both acceptors active case. Looks still cleaner to me then initial implementation with two separate methods/etc. If you think it is overkill, lmk I will roll it back :)